src/actions/channels.js

```diff
@@ -711,14 +714,20 @@ export function viewChannel(channelId: string, prevChannelId: string = ''): Acti
         const {myMembers} = getState().entities.channels;
         const member = myMembers[channelId];
         if (member) {
+            if (isManuallyUnread(channelId, getState())) {
```

Comment:
what if you do `const state = getState()` and then just pass the state object instead of calling the function every time?

Reply:
I can only change it in one place ,all the others has an `await Client4.<some api call>` in the middle so there might be some changed in the state by that time (most likely not, but there is the option)